### PR TITLE
ENH color changes for y6

### PR DESCRIPTION
**Pull request by beckermr**
This PR has changes for Y6 color dependence in our data products.

### CODE REVIEW

File: meds/maker.py [hunk 1]
```diff
@@ -1373,16 +1400,19 @@ def _load_config(self, config):
...
-            psf_data[file_id].get_rec(row, col)
-            for file_id, row, col in zip(file_ids, rows, cols)
-        ],
-        output_path,
-    )
+    psfs = []
+    for file_id, row, col, color in zip(file_ids, rows, cols, colors):
+        try:
+            psf = psf_data[file_id].get_rec(row, col, color=color)
+        except TypeError:
```
Comment (esheldon) [inline meds/maker.py:1410]: if the user sent colors but the function doesn't accept colors, we should be crashing.

File: meds/maker.py [hunk 2]
```diff
@@ -1190,7 +1214,10 @@ def _set_psf_layout(self):
                 if hasattr(p, "get_rec_shape"):
                     psf_shape = p.get_rec_shape(row, col)
                 else:
-                    psf_shape = p.get_rec(row, col).shape
+                    try:
+                        psf_shape = p.get_rec(row, col, color=color).shape
+                    except TypeError:
```
Comment (esheldon) [inline meds/maker.py:1219]: Here also we should not allow this

File: meds/maker.py [hunk 3]
```diff
@@ -396,7 +408,10 @@ def _write_psf_cutouts_serial(self):
                 col = obj_data["orig_col"][iobj, icut]
                 start_row = obj_data["psf_start_row"][iobj, icut]
 
-                psfim = psf_data[file_id].get_rec(row, col)
+                try:
+                    psfim = psf_data[file_id].get_rec(row, col, color=color)
+                except TypeError:
```
Comment (esheldon) [inline meds/maker.py:413]: and here

**Review by esheldon — CHANGES_REQUESTED** [review]

File: meds/maker.py [hunk 4]
```diff
@@ -1373,26 +1396,26 @@ def _load_config(self, config):
...
-
     if color is not None:
-        try:
-            res = wcs.sky2image(ra, dec, color=color)
-        except TypeError:
-            res = wcs.sky2image(ra, dec)
+        res = wcs.sky2image(ra, dec, color=color)
     else:
         res = wcs.sky2image(ra, dec)
+
```
Comment (beckermr) [inline meds/maker.py:1420]: I changed the TypeError handling per your request. Note that I was following the pattern here in the WCS handling. I've changed that pattern here too.

**Comment by beckermr** [comment]
Ahhhhh, we have to keep the try except blocks because the coadd quantities don't always accept a color but we pass it anyways.

**Comment by beckermr** [comment]
bump here @esheldon

**Comment by esheldon** [comment]
> Ahhhhh, we have to keep the try except blocks because the coadd quantities don't always accept a color but we pass it anyways.

I think the right thing to do would be to identify when that situation occurs and deal with it separately, to avoid user error in this case.

If you think this is just not worth it, I will accept your judgement.

**Review by esheldon — COMMENTED** [review]
See comment in conversation thread

**Comment by beckermr** [comment]
OK @esheldon ready for another look!

**Review by esheldon — APPROVED** [review]

**Pull request merged**